pydarn/plotting/fan.py

```diff
@@ -185,11 +183,28 @@ def plot_fan(cls, dmap_data: List[dict], ax=None,
         	date = scan_time
 
         # Plot FOV outline
-        if ranges is None:
-            ranges = [0, dmap_data[0]['nrang']]
+        if ranges == [] or ranges is None:
+            try:
+                # If not given, get ranges from data file
+                ranges = [0, dmap_data[0]['nrang']]
+            except KeyError:
+                # Otherwise, default to [0,75]
+                ranges = [0,75]
+        
+        # Get rsep and frang from data unless not there then take defaults
+        try:
+            frang = dmap_data[0]['frang']
+        except KeyError:
+            frang = 180
+        
+        try:
+            rsep = dmap_data[0]['rsep']
+        except:
+            rsep = 45
```

Comment:
Can you add some comments on these defaults and these are not in hardware files? 

They keyError should trigger but I am okay with being safe than sorry

Reply:
frang and rsep are not in hardware files as they can change between control programs, but are generally in the data unless it's a partial record. There was that whole thing when we first did the ranges keyword where we ended up just hard coding the max ranges value into the SuperdarnRadars class so there was something to fall back on.
The other option here instead of using a default value, or hardcoding more defaults in, to keep the code running would be to throw an error saying that there's no frang or rsep value in the record and so we can't plot it accurately, or throw a warning so the user knows that they could be slightly wrong?

Reply:
Nope I am cool with it, do you mind just adding the comment on the values chosen though so we don't have to remember why we did those. I get 45 km makes sense just pray it didn't run on 15 km mode lol